src/element/newElement.ts

```diff
@@ -246,11 +246,17 @@ export const updateTextElement = (
     originalText,
   }: { text: string; isDeleted?: boolean; originalText: string },
 
-  updateDimensions: boolean,
+  isSubmit: boolean,
 ): ExcalidrawTextElement => {
-  const dimensions = updateDimensions
-    ? getAdjustedDimensions(element, text)
-    : undefined;
+  const boundToContainer = isBoundToContainer(element);
+
+  // Don't update dimensions and text value for bounded text unless submitted
+  const dimensions =
+    boundToContainer && !isSubmit
+      ? undefined
+      : getAdjustedDimensions(element, text);
+
+  text = boundToContainer && !isSubmit ? element.text : text;
```

Comment:
why updating `text` to stale state while editing bound text element?

Reply:
I don't want to update text until submitted so that before submit its possible to compare old text and new text and check if mutation is needed as done in line 457 in `textwysiswg.tsx`

Reply:
This creates complexity and exposing details all over the codebase that shouldn't be exposed. Also, it breaks collaboration (latest value is not synced to remote clients while editing), and may break other things (now or in the future). Lastly, it's not clear from the code what this does at all, so would require extensive comments and such.

In any case, we can just keep the original text as a local state within the wysiwyg editor closure, no?